examples/decoupled/repeat_model.py

```diff
@@ -0,0 +1,206 @@
+                serve this model""".format(args['model_name']))
+
+        # Get OUT configuration
+        out_config = pb_utils.get_output_config_by_name(model_config, "OUT")
+
+        # Get IDX configuration
+        idx_config = pb_utils.get_output_config_by_name(model_config, "IDX")
+
+        # Convert Triton types to numpy types
+        self.out_dtype = pb_utils.triton_string_to_numpy(
+            out_config['data_type'])
+        self.idx_dtype = pb_utils.triton_string_to_numpy(
+            idx_config['data_type'])
+
+        # To keep track of response threads so that we can delay
+        # the finalizing the model until all response threads
+        # have completed.
+        self.inflight_thread_count = 0
+        self.inflight_thread_count_lck = threading.Lock()
+
+    def execute(self, requests):
+        """`execute` MUST be implemented in every Python model. `execute`
+        function receives a list of pb_utils.InferenceRequest as the only
+        argument. This function is called when an inference request is made
+        for this model. The request.get_response_sender() must be used to
+        get an InferenceResponseSender object associated with the request.
+        Use the InferenceResponseSender.send() and InferenceResponseSender.close()
+        calls to send responses and indicate no responses will be sent for
+        the corresponding request respectively. If there is an error, you can
+        set the error argument when creating a pb_utils.InferenceResponse
+
+        Parameters
+        ----------
+        requests : list
+          A list of pb_utils.InferenceRequest
+
+        Returns
+        -------
+        None
+        """
+
+        # This model does not support batching, so 'request_count' should always be 1.
+        if len(requests) != 1:
+            raise pb_utils.TritonModelException("unsupported batch size " +
+                                                len(requests))
+
+        # Start a separate thread to send the responses for the request.
+        thread = threading.Thread(
+            target=response_thread,
+            args=(self, requests[0].get_response_sender(),
+                  pb_utils.get_input_tensor_by_name(requests[0],
+                                                    'IN').as_numpy(),
+                  pb_utils.get_input_tensor_by_name(requests[0],
+                                                    'DELAY').as_numpy()))
+        thread.daemon = True
+
+        with self.inflight_thread_count_lck:
+            self.inflight_thread_count += 1
+
+        thread.start()
+
+        # Read WAIT input for wait time, then return so that Triton can call execute again with
+        # another request.
+        wait_input = pb_utils.get_input_tensor_by_name(requests[0],
+                                                       'WAIT').as_numpy()
+        time.sleep(wait_input[0] / 1000)
+
+        return None
```

Comment:
Can you add a comment above this sentence and contrast this with the non-decoupled API mode, where the has to be a list of inference response objects here?